cmd/api.go

```diff
@@ -0,0 +1,649 @@
+	if err != nil {
+		helpers.WriteErrorResponse(http.StatusBadRequest, err, w)
+		return
+	}
+
+	height, err := helpers.ParseHeightFromRequest(r, chain)
+	if err != nil {
+		helpers.WriteErrorResponse(http.StatusBadRequest, err, w)
+		return
+	}
+
+	if err = chain.AddPath(dcli, dcon, vars["chan-id"], vars["port-id"], dord); err != nil {
+		helpers.WriteErrorResponse(http.StatusInternalServerError, err, w)
+		return
+	}
+
+	res, err := chain.QueryChannel(height)
+	if err != nil {
+		helpers.WriteErrorResponse(http.StatusInternalServerError, err, w)
+		return
+	}
+	helpers.SuccessProtoResponse(http.StatusOK, chain, res, w)
+}
+
+// QueryChannelsHandler handles the route
+func QueryChannelsHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	chain, err := config.Chains.Get(vars["chain-id"])
+	if err != nil {
+		helpers.WriteErrorResponse(http.StatusBadRequest, err, w)
+		return
+	}
+
+	offset, limit, err := helpers.ParsePaginationParams(r)
+	if err != nil {
+		helpers.WriteErrorResponse(http.StatusBadRequest, err, w)
+		return
+	}
+
+	res, err := chain.QueryChannels(offset, limit)
+	if err != nil {
+		helpers.WriteErrorResponse(http.StatusInternalServerError, err, w)
+		return
+	}
+	helpers.SuccessProtoResponse(http.StatusOK, chain, res, w)
+}
+
+// QueryConnectionChannelsHandler handles the route
+func QueryConnectionChannelsHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	chain, err := config.Chains.Get(vars["chain-id"])
+	if err != nil {
+		helpers.WriteErrorResponse(http.StatusBadRequest, err, w)
+		return
+	}
+
+	if err = chain.AddPath(dcli, vars["conn-id"], dcha, dpor, dord); err != nil {
+		helpers.WriteErrorResponse(http.StatusInternalServerError, err, w)
+		return
+	}
+
+	offset, limit, err := helpers.ParsePaginationParams(r)
+	if err != nil {
+		helpers.WriteErrorResponse(http.StatusInternalServerError, err, w)
+		return
+	}
+
+	res, err := chain.QueryConnectionChannels(vars["conn-id"], offset, limit)
+	if err != nil {
+		helpers.WriteErrorResponse(http.StatusInternalServerError, err, w)
+		return
+	}
+	helpers.SuccessProtoResponse(http.StatusOK, chain, res, w)
+}
+
+// QueryIBCDenomsHandler handles the route
+func QueryIBCDenomsHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	chain, err := config.Chains.Get(vars["chain-id"])
+	if err != nil {
+		helpers.WriteErrorResponse(http.StatusBadRequest, err, w)
+		return
+	}
+
+	h, err := chain.QueryLatestHeight()
+	if err != nil {
+		helpers.WriteErrorResponse(http.StatusInternalServerError, err, w)
+		return
+	}
+
+	res, err := chain.QueryDenomTraces(0, 1000, h)
+	if err != nil {
+		helpers.WriteErrorResponse(http.StatusInternalServerError, err, w)
+		return
+	}
+	helpers.SuccessProtoResponse(http.StatusOK, chain, res, w)
+}
+
+// PostRelayerListenHandler returns a handler for a listener that can listen on many IBC paths
+func PostRelayerListenHandler(sm *ServicesManager) func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Does this work? That would a step change in the relayer capabilities.

Reply:
We need to do some rework for this api. We can implement all TODOs related to `PostRelayerListenHandler` in separate PR.

Reply:
could you open an issue to track this work?